src/lib/common/Makefile

```diff
@@ -53,8 +53,6 @@ obj-core-$(PLATFORM_CONTIKI) += \
 obj-core-$(PLATFORM_LINUX) += \
     sol-platform-linux-common.o \
     sol-board-detect.o
-obj-core-$(PLATFORM_LINUX)-extra-cflags += $(SYSTEMD_CFLAGS)
-obj-core-$(PLATFORM_LINUX)-extra-ldflags += $(SYSTEMD_LDFLAGS)
```

Comment:
the point is, if you disable systemd and enable sd-bus the target that will link with systemd is the SOL_BUS isn't ? Or are you saying that if disable systemd it still exists obj-core-$(PLATFORM_SYSTEMD) ?